runtime-environment-spec/polkadot_re_spec.tm

```diff
@@ -1937,7 +1937,132 @@
+    between storage key and extrinsics (<reference|defn-storage-key-to-extrinsics>)>|<cell|<em|KeyIndex>
+    (<reference|defn-key-index>)>>|<row|<cell|2>|<cell|Mapping between
+    storage key and blocks (<reference|defn-storage-key-to-blocks>)>|<cell|<em|KeyIndex>
+    (<reference|defn-key-index>)>>|<row|<cell|3>|<cell|Mapping between
+    storage key and child changes Trie (<reference|defn-storage-key-to-child-tries>)>|<cell|<em|KeyIndex>
+    (<reference|defn-key-index>)>>>>>
+
+    \;
+  <|big-table>
+    <label|table-changes-trie-key-types>Varying datatype of possible mappings
+    in the Changes Trie
+  </big-table>
+
+  <strong|Note>: the key used for the Changes Trie must be the <em|varying
+  datatype>, not the individual, appended <em|KeyIndex>. Unlike the default
+  encoding for varying data types, this structure starts its indexing at
+  <verbatim|1>.
+
+  <\definition>
+    <label|defn-key-index>The <strong|Key Index> is a tuple containing the
+    following information:
+
+    <\big-table|<tabular|<tformat|<cwith|2|2|1|1|cell-lborder|0ln>|<cwith|2|2|3|3|cell-rborder|0ln>|<cwith|3|3|1|-1|cell-tborder|1ln>|<cwith|2|2|1|-1|cell-bborder|1ln>|<cwith|3|3|1|-1|cell-bborder|1ln>|<cwith|3|3|1|1|cell-lborder|0ln>|<cwith|3|3|3|3|cell-rborder|0ln>|<cwith|1|1|1|-1|cell-tborder|1ln>|<cwith|1|1|1|-1|cell-bborder|1ln>|<cwith|2|2|1|-1|cell-tborder|1ln>|<cwith|1|1|1|1|cell-lborder|0ln>|<cwith|1|1|3|3|cell-rborder|0ln>|<table|<row|<cell|<strong|Name>>|<cell|<strong|Description>>|<cell|<strong|Type>>>|<row|<cell|Block>|<cell|Block
+    number at which this key has been inserted in the trie>|<cell|Unsigned
+    32-bit integer>>|<row|<cell|Key>|<cell|The changed key>|<cell|Byte
+    array>>>>>>
+      Key structure inserted in the Changes Trie
+    </big-table>
+  </definition>
+
+  <\definition>
+    <label|defn-storage-key-to-extrinsics>The <strong|storage key to
+    extrinsics mappings> track any changes which occure in an individual
+    block. The key is inserted into the Trie, where its value is a SCALE
+    encoded array containing the indices of the extrinsics that caused any
+    changes to the key. The indices are unsigned 32-bit integers and it's
+    values depend on the order in which each extrinsic is included in the
+    block. Additionally, a separate Changes Trie is created for each child
+    storage where changes occure.
+
+    \;
+
+    Keys that were modified outside of extrinsics, such as changes made by
+    the Runtime directly, have a <verbatim|0xffffffff> value as its indice.
+
+    \;
+
+    Those entries are generated when the Runtime calls
```

Comment:
It's not quite clear to me what is meant by generation of the entries. I may try to guess or ask you, but as it's not quite clear now, maybe you might clarify it a bit.